src/ngraph/op/fused/clamp.cpp

```diff
@@ -18,12 +18,46 @@
 #include "ngraph/builder/make_constant.hpp"
 #include "ngraph/op/maximum.hpp"
 #include "ngraph/op/minimum.hpp"
+#include "ngraph/runtime/reference/clamp.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::Clamp::type_info;
 
+namespace
+{
+    template <element::Type_t ET>
+    bool evaluate(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        runtime::reference::clamp(
+            arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), min, max, count);
+        return true;
+    }
+
+    bool evaluate_clamp(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        bool rc = true;
+        switch (arg->get_element_type())
+        {
+            TYPE_CASE(f32)(arg, out, min, max, count);
```

Comment:
The spec doesn't restrict the input types, just the min/max types.

Reply:
OK, will fix.  This will force me to static_cast the argument to a `double` for comparison with min/max.  The semantics of this op are strange.  Why not have min/max be of the same type as the argument?

Reply:
They are attributes; we don't have variant types that they could be.

Reply:
Added support for all types.